gammapy/cube/fit.py

```diff
@@ -51,56 +45,60 @@ def __init__(
         if mask is not None and mask.data.dtype != np.dtype("bool"):
             raise ValueError("mask data must have dtype bool")
 
-        self._model = model
+        self.model = model
         self.counts = counts
         self.exposure = exposure
-        self.background = background
         self.mask = mask
         self.psf = psf
         self.edisp = edisp
+        self.background_model = background_model
+        if background_model:
+            self.parameters = Parameters(
+                self.model.parameters.parameters
+                + self.background_model.parameters.parameters
+            )
+        else:
+            self.parameters = Parameters(self.model.parameters.parameters)
 
         self.evaluator = MapEvaluator(
-            model=self._model,
-            exposure=exposure,
-            background=self.background,
-            psf=self.psf,
-            edisp=self.edisp,
-            background_model=background_model,
+            model=self.model, exposure=exposure, psf=self.psf, edisp=self.edisp
         )
 
     @property
-    def stat(self):
+    def npred(self):
+        """Returns npred map (model + background)"""
+        model_npred = self.evaluator.compute_npred()
+        back_npred = self.background_model.evaluate()
+        total_npred = model_npred.data + back_npred.data
+        return back_npred.copy(data=total_npred)
+        # TODO: return model_npred + back_npred
+        # There is some bug: edisp.e_reco.unit is dimensionless
+        # thus map arithmetic does not work.
+
+    def likelihood_per_bin(self):
         """Likelihood per bin given the current model parameters"""
-        npred = self.evaluator.compute_npred()
-        return cash(n_on=self.counts.data, mu_on=npred)
+        return cash(n_on=self.counts.data, mu_on=self.npred.data)
 
-    def total_stat(self, parameters):
+    def likelihood(self, parameters):
```

Comment:
Passing `parameters` here is unnecessary, but this is because there are too many callers to `total_stat` in `gammapy.utils`. I would prefer to clean up this caller in a follow up once this PR is merged (also needs cleanup in `SpectrumFit` and `FluxPointDataset`.

Reply:
Thanks! Ok to merge from my side then